### DISCUSSION TITLE
How can I clear the current triples in qEndpoint and replace with the new ones?

### DISCUSSION THREAD

**Discussion by bohuizhang** [the-qa-company/qEndpoint#488] [Q&A] [answered]
I am using the qEndpoint for loading and querying local knowledge graphs. Could you advise on how to clear the currently loaded/indexed triples and replace them with new ones? I attempted to use the 'DELETE' command, but the newly loaded triples are not being returned in queries.

**Comment by ate47** [comment, marked as answer]
Weirdly CLEAR ALL doesn't work, I'll have to look at it. This one works:

```sql
DELETE WHERE { ?s ?p ?o }
```

But qEndpoint is more in the spirit, we index/load a big dataset (HDT) and then we add/remove parts from it. If you want to fully erase the dataset, you can stop the endpoint, delete the `hdt-store` and `native-store` directories in your qEndpoint installation and restart it.

**Reply by bohuizhang** [reply]
The `DELETE` command doesn't seem to be working properly. After executing the `DELETE` command and loading new data, the newly loaded triples are not queryable, even after restarting the endpoint. While the alternative approach (deleting the `hdt-store` and `native-store` directories) works!

**Reply by ate47** [reply]
I noticed that in #490, for now if you want to load another dataset you need to do what I said with deleting the previous dataset.
